### ISSUE TITLE
Stackoverflow on attempting `dumpEntities` command with CoreSampleGameplay

### ISSUE THREAD

**Issue by Cervator** [MovingBlocks/Terasology#3877]
### What you were trying to do

Use the `dumpEntities` command to get a text file with all active entities written to my workspace for troubleshooting. 

### What actually happened

Stackoverflow exception. See log snippet below.

May relate to #3739 - perhaps we have a prefab with a circular reference / dependency in the hundred of so entities that get registered in a clean new game?

Need to find and fix the issue so the command is useful again. Fixing the underlying architecture to better identify and bypass bad prefabs would be better dealt with via #3739 although fixing both at once would be best!

### How to reproduce

* Run the game
* Start a new world with CoreSampleGameplay defaults
* Execute `dumpEntities` in the game console

### Log details and game version

```
15:02:34.567 [main] ERROR o.t.logic.console.ConsoleImpl - An error occurred while executing a command
org.terasology.logic.console.commandSystem.exceptions.CommandExecutionException: java.lang.StackOverflowError
	at org.terasology.logic.console.commandSystem.AbstractCommand.execute(AbstractCommand.java:242)
	at org.terasology.logic.console.ConsoleImpl.execute(ConsoleImpl.java:264)
	at org.terasology.logic.console.ConsoleImpl.execute(ConsoleImpl.java:228)
	at org.terasology.logic.console.ui.ConsoleScreen.lambda$initialise$0(ConsoleScreen.java:79)
	at org.terasology.rendering.nui.widgets.UIText.onKeyEvent(UIText.java:468)
	at org.terasology.logic.console.ui.UICommandEntry.onKeyEvent(UICommandEntry.java:83)
	at org.terasology.rendering.nui.internal.NUIManagerInternal.keyEvent(NUIManagerInternal.java:648)
	at org.terasology.rendering.nui.internal.NUIManagerInternalMethodAccess.invoke(Unknown Source)
	at org.terasology.entitySystem.event.internal.EventSystemImpl$ByteCodeEventHandlerInfo.invoke(EventSystemImpl.java:531)
	at org.terasology.entitySystem.event.internal.EventSystemImpl.sendConsumableEvent(EventSystemImpl.java:301)
	at org.terasology.entitySystem.event.internal.EventSystemImpl.send(EventSystemImpl.java:280)
	at org.terasology.entitySystem.entity.internal.BaseEntityRef.send(BaseEntityRef.java:205)
	at org.terasology.input.InputSystem.send(InputSystem.java:503)
	at org.terasology.input.InputSystem.sendKeyEvent(InputSystem.java:447)
	at org.terasology.input.InputSystem.processKeyboardInput(InputSystem.java:382)
	at org.terasology.input.InputSystem.update(InputSystem.java:135)
	at org.terasology.engine.modes.StateIngame.handleInput(StateIngame.java:192)
	at org.terasology.engine.subsystem.lwjgl.LwjglInput.postUpdate(LwjglInput.java:55)
	at org.terasology.engine.TerasologyEngine.tick(TerasologyEngine.java:471)
	at org.terasology.engine.TerasologyEngine.mainLoop(TerasologyEngine.java:425)
	at org.terasology.engine.TerasologyEngine.run(TerasologyEngine.java:401)
	at org.terasology.engine.Terasology.main(Terasology.java:162)
Caused by: java.lang.StackOverflowError: null
	at sun.reflect.misc.ReflectUtil.checkPackageAccess(ReflectUtil.java:188)
	at sun.reflect.misc.ReflectUtil.checkPackageAccess(ReflectUtil.java:164)
	at sun.reflect.generics.reflectiveObjects.TypeVariableImpl.getGenericDeclaration(TypeVariableImpl.java:164)
	at sun.reflect.generics.reflectiveObjects.TypeVariableImpl.equals(TypeVariableImpl.java:189)
	at java.util.Arrays$ArrayList.indexOf(Arrays.java:3860)
	at org.terasology.utilities.ReflectionUtil.lambda$cascadeTypeVariableDeclarationToSupertypes$4(ReflectionUtil.java:444)
	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:267)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.stream.Streams$StreamBuilderImpl.forEachRemaining(Streams.java:419)
	at java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:742)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.StreamSpliterators$WrappingSpliterator.forEachRemaining(StreamSpliterators.java:312)
	at java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:743)
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:580)
	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:270)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.stream.Streams$StreamBuilderImpl.tryAdvance(Streams.java:405)
	at java.util.stream.Streams$ConcatSpliterator.tryAdvance(Streams.java:728)
	at java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:294)
	at java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:206)
	at java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:161)
	at java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:300)
	at java.util.stream.Streams$ConcatSpliterator.tryAdvance(Streams.java:731)
	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:126)
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:498)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:485)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:152)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.findAny(ReferencePipeline.java:469)
	at org.terasology.utilities.ReflectionUtil.resolveTypeVariable(ReflectionUtil.java:402)
	at org.terasology.utilities.ReflectionUtil.resolveType(ReflectionUtil.java:306)
	at org.terasology.utilities.ReflectionUtil.resolveTypes(ReflectionUtil.java:387)
	at org.terasology.utilities.ReflectionUtil.resolveType(ReflectionUtil.java:331)
	at org.terasology.utilities.ReflectionUtil.parameterizeandResolveRawType(ReflectionUtil.java:510)
	at org.terasology.persistence.typeHandling.coreTypes.RuntimeDelegatingTypeHandler.getRuntimeTypeIfMoreSpecific(RuntimeDelegatingTypeHandler.java:170)
	at org.terasology.persistence.typeHandling.coreTypes.RuntimeDelegatingTypeHandler.serializeNonNull(RuntimeDelegatingTypeHandler.java:76)
	at org.terasology.persistence.typeHandling.TypeHandler.serialize(TypeHandler.java:49)
	at org.terasology.persistence.typeHandling.coreTypes.ObjectFieldMapTypeHandler.serializeNonNull(ObjectFieldMapTypeHandler.java:71)
	at org.terasology.persistence.typeHandling.TypeHandler.serialize(TypeHandler.java:49)
	at org.terasology.persistence.typeHandling.coreTypes.RuntimeDelegatingTypeHandler.serializeNonNull(RuntimeDelegatingTypeHandler.java:131)
	at org.terasology.persistence.typeHandling.TypeHandler.serialize(TypeHandler.java:49)
```

**Comment by pollend** [comment]
I added 
```

16:24:38.837 [Thread-3] DEBUG c.j.d.entities.pipe.UnixPipe - Received packet: Pkt:FRAME{"evt":null,"data":{"assets":{"large_image":"515746954660151306"},"timestamps":{"start":1586129069000},"name":"Terasology","details":"Name: Player70857","state":"Solo | Game 7","secrets":{},"application_id":"515274721080639504"},"cmd":"SET_ACTIVITY","nonce":"a36f0a9b-30f7-45b2-beb8-baba4202110c"}
16:24:42.043 [main] ERROR o.t.p.s.WorldSerializerImpl - failed to serialize prefab: engine:camera {}
java.lang.StackOverflowError: null
	at org.terasology.utilities.ReflectionUtil.lambda$cascadeTypeVariableDeclarationToSupertypes$4(ReflectionUtil.java:444)
	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:269)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:958)
	at java.util.stream.Streams$ConcatSpliterator.tryAdvance(Streams.java:731)
	at java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:295)
	at java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:207)
	at java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:162)
	at java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:301)
	at java.util.stream.Streams$ConcatSpliterator.tryAdvance(Streams.java:731)
	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.stream.Streams$StreamBuilderImpl.tryAdvance(Streams.java:405)
	at java.util.stream.Streams$ConcatSpliterator.tryAdvance(Streams.java:728)
	at java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:295)
	at java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:207)
	at java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:162)
	at java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:301)
	at java.util.stream.Streams$ConcatSpliterator.tryAdvance(Streams.java:731)
	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.stream.Streams$StreamBuilderImpl.tryAdvance(Streams.java:405)
	at java.util.stream.Streams$ConcatSpliterator.tryAdvance(Streams.java:728)
	at java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:295)
	at java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:207)
	at java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:162)
	at java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:301)
	at java.util.stream.Streams$ConcatSpliterator.tryAdvance(Streams.java:731)
	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:126)
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:499)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:486)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:152)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.findAny(ReferencePipeline.java:536)
	at org.terasology.utilities.ReflectionUtil.resolveTypeVariable(ReflectionUtil.java:402)
	at org.terasology.utilities.ReflectionUtil.resolveType(ReflectionUtil.java:306)
	at org.terasology.utilities.ReflectionUtil.resolveTypes(ReflectionUtil.java:387)
	at org.terasology.utilities.ReflectionUtil.resolveType(ReflectionUtil.java:331)
	at org.terasology.utilities.ReflectionUtil.parameterizeandResolveRawType(ReflectionUtil.java:510)
	at org.terasology.persistence.typeHandling.coreTypes.RuntimeDelegatingTypeHandler.getRuntimeTypeIfMoreSpecific(RuntimeDelegatingTypeHandler.java:170)
	at org.terasology.persistence.typeHandling.coreTypes.RuntimeDelegatingTypeHandler.serializeNonNull(RuntimeDelegatingTypeHandler.java:76)
	at org.terasology.persistence.typeHandling.TypeHandler.serialize(TypeHandler.java:49)
	at org.terasology.persistence.typeHandling.coreTypes.ObjectFieldMapTypeHandler.serializeNonNull(ObjectFieldMapTypeHandler.java:71)
	at org.terasology.persistence.typeHandling.TypeHandler.serialize(TypeHandler.java:49)
	at org.terasology.persistence.typeHandling.coreTypes.RuntimeDelegatingTypeHandler.serializeNonNull(RuntimeDelegatingTypeHandler.java:131)
	at org.terasology.persistence.typeHandling.TypeHandler.serialize(TypeHandler.java:49)
	at org.terasology.persistence.typeHandling.coreTypes.ObjectFieldMapTypeHandler.serializeNonNull(ObjectFieldMapTypeHandler.java:71)
	at org.terasology.persistence.typeHandling.TypeHandler.serialize(TypeHandler.java:49)
	at org.terasology.persistence.typeHandling.coreTypes.RuntimeDelegatingTypeHandler.serializeNonNull(RuntimeDelegatingTypeHandler.java:131)
	at org.terasology.persistence.typeHandling.TypeHandler.serialize(TypeHandler.java:49)
	at org.terasology.persistence.typeHandling.coreTypes.ObjectFieldMapTypeHandler.serializeNonNull(ObjectFieldMapTypeHandler.java:71)
	at org.terasology.persistence.typeHandling.TypeHandler.serialize(TypeHandler.java:49)
	at org.terasology.persistence.typeHandling.coreTypes.RuntimeDelegatingTypeHandler.serializeNonNull(RuntimeDelegatingTypeHandler.java:131)
	at org.terasology.persistence.typeHandling.TypeHandler.serialize(TypeHandler.java:49)
	at org.terasology.persistence.typeHandling.coreTypes.ObjectFieldMapTypeHandler.serializeNonNull(ObjectFieldMapTypeHandler.java:71)
	at org.terasology.persistence.typeHandling.TypeHandler.serialize(TypeHandler.java:49)
	at org.terasology.persistence.typeHandling.coreTypes.RuntimeDelegatingTypeHandler.serializeNonNull(RuntimeDelegatingTypeHandler.java:131)
	at org.terasology.persistence.typeHandling.TypeHandler.serialize(TypeHandler.java:49)
	at org.terasology.persistence.typeHandling.coreTypes.ObjectFieldMapTypeHandler.serializeNonNull(ObjectFieldMapTypeHandler.java:71)
	at org.terasology.persistence.typeHandling.TypeHandler.serialize(TypeHandler.java:49)
	at org.terasology.persistence.typeHandling.coreTypes.RuntimeDelegatingTypeHandler.serializeNonNull(RuntimeDelegatingTypeHandler.java:131)
	at org.terasology.persistence.typeHandling.TypeHandler.serialize(TypeHandler.java:49)
...
```

**Comment by Cervator** [comment]
Closing as fixed by #3879 (am mildly surprised at the fix though I will say, but ok!)
